common/authentication/aws/x509.go

```diff
@@ -0,0 +1,479 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/tls"
+	cryptoX509 "crypto/x509"
+	"errors"
+	"fmt"
+	"net/http"
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	awssh "github.com/aws/rolesanywhere-credential-helper/aws_signing_helper"
+	"github.com/aws/rolesanywhere-credential-helper/rolesanywhere"
+	"github.com/aws/rolesanywhere-credential-helper/rolesanywhere/rolesanywhereiface"
+
+	cryptopem "github.com/dapr/kit/crypto/pem"
+	spiffecontext "github.com/dapr/kit/crypto/spiffe/context"
+	"github.com/dapr/kit/logger"
+	kitmd "github.com/dapr/kit/metadata"
+	"github.com/dapr/kit/ptr"
+)
+
+type x509 struct {
+	mu sync.RWMutex
+
+	wg      sync.WaitGroup
+	closeCh chan struct{}
+
+	logger              logger.Logger
+	Clients             *Clients
+	rolesAnywhereClient rolesanywhereiface.RolesAnywhereAPI // this is so we can mock it in tests
+	session             *session.Session
+	Cfg                 *aws.Config
+
+	chainPEM []byte
+	keyPEM   []byte
+
+	region          *string
+	TrustProfileArn *string        `json:"trustProfileArn" mapstructure:"trustProfileArn"`
+	TrustAnchorArn  *string        `json:"trustAnchorArn" mapstructure:"trustAnchorArn"`
+	AssumeRoleArn   *string        `json:"assumeRoleArn" mapstructure:"assumeRoleArn"`
+	SessionDuration *time.Duration `json:"sessionDuration" mapstructure:"sessionDuration"`
```

Comment:
Why are these exported? Options which are parsed shouldn't belong to this struct.